src/pond/lib/timeseries.js

```diff
@@ -385,7 +388,7 @@ class TimeSeries {
      */
     slice(begin, end) {
         const sliced = this._collection.slice(begin, end);
-        return this.setCollection(sliced);
+        return this.setCollection(sliced, true);
```

Comment:
Not sure on the JS convention here, but usually if you use a kwarg in python it's better to use the name when calling the function. This makes the position of the kwarg implicitly important, at least in Python.  Again, not sure the common use case in JS.